go/worker/registration/worker.go

```diff
@@ -656,6 +672,7 @@ func (w *Worker) registerNode(epoch epochtime.EpochTime, hook RegisterNodeHook)
 	}
 
 	w.logger.Info("node registered with the registry")
+	workerNodeRegistered.Set(1.0)
```

Comment:
What about resetting to zero in case the registration transaction failed (e.g., a few lines above, but basically any of the error cases)? I would move setting this metric to where `registerNode` is called, so that you can check the error code (and if there is an error you need to reset it to zero).